bin/p2-preparer/main.go

```diff
@@ -44,6 +44,12 @@ func main() {
 		defer statusServer.Close()
 	}
 
+	if preparerConfig.RequireFile != "" {
```

Comment:
In my understanding, it was so that p2-exec would not need to have the required file hard-coded. That goal has been achieved in this PR, I believe: in this PR, p2-exec doesn't have the required file hard-coded, and it only knows about it by being passed the --required-file flag. The p2-preparer writes the appropriate servicebuilder/runit files so that when p2-exec runs, it has the right --required-file flag.

This was merely a suspicion that we wouldn't want to be too opinionated in open-source code. If we didn't care, it might have been acceptable to have the required file be a constant in p2-exec, and have p2-preparer be completely unaware of it. It seems that the way this PR has it (have it in p2-preparer's config) is the way we decided on, though.